supervisor/supervisor.go

```diff
@@ -291,6 +292,14 @@ func (s *Supervisor) Machine() Machine {
 
 // SendTask sends the provided event the the supervisors main event loop
 func (s *Supervisor) SendTask(evt Task) {
+	ctxVal := fmt.Sprintf("%v", evt.Ctx())
+	if evt.Ctx().Done() == nil && ctxVal == "context.Background" {
```

Comment:
this is digging into the context package - I'm unsure how this is helping, where in containerd we don't have a ctx which is not background? 🤔

Reply:
This was my struggle.  Granted I know not a whole lot about containerd.  My issue was I was trying to determine if the task/event was still active or not, and if not, to kill it.  The test is basically to send an unterminated cat command in a setup like this:

Change the defaultBufferSize value in supervisor.go from 2048 to 128.
```
# docker create --name=test registry.access.redhat.com/rhel7 sleep 365d
# docker start test
# for i in `seq 1 129`; do docker exec -d test cat; done
# kill -STOP {dockerd PID} {containerd PID}
# pkill cat
# kill -CONT {dockerd PID} ; sleep 30 ; systemctl status docker
```
then
``` 
# journalctl -u docker | grep "libcontainerd: failed to receive event from containerd: rpc error: code = 12 desc = unknown service types.API" | wc -l
```
Returns thousands of hits.  If you dump core, the process is just hanging out at the select statement, in this file line 303 below.

What I was seeing prior was the evt->ctx before I did all the kills had a huge amount of data in it, which started with:
```
evt.Ctx [context.TODO.WithCancel.WithValue(peer.peerKey{}....]
evt.Ctx.Done() [0xc00002dce0]
```
After I the kill steps completed, then I had values that looked like:
```
evt.Ctx() [context.Background]
evt.Ctx.Done() [<nil>]
```
This all seemed to work for my testing for this particular BZ, but I was and still am concerned, that the evt variable goes into this state not only for this particular instance but also when things were good and it wouldn't be appropriate to not send the task.

Reply:
Right, that's an interesting point also. I still think we shouldn't rely in this check tho. Maybe the bug is hiding where this evt object gets manipulated? So instead of doing this, we got to the root and actually stop this event altogether maybe